### ISSUE TITLE
Enhancement: Dynamic template enhancement in Dialog Service

### ISSUE THREAD

**Issue by jedd-ahyoung** [aurelia/dialog#249]
I'm looking for a way to wrap an existing view in a custom element that uses named slots, without having to specify the custom element inside of the view itself. The idea is to allow a view to use slots to separate its content, without embedding dialog-specific elements in the view itself, so that it can be reused easily inside and outside of dialogs.

I've come up with a very primitive implementation for an example of a Bootstrap dialog renderer; it dynamically wraps an existing view with a custom element and calls `TemplateEngine.enhance` on the created element. The code is here: https://github.com/jedd-ahyoung/bootstrap-dialog-renderer (demo at http://jedd-ahyoung.com/bootstrap-dialog-renderer/).

The problem with the approach linked to above is that the enhancement happens too late in the pipeline; because the view and viewmodel have already been composed, calling `TemplateEngine.enhance` on an element that has already been enhanced results in strange bugs (like trigger events being registered twice, etc).

I'd like to propose enhancing the pipeline in DialogService so that it can wrap a view in an element of the user's choice before the viewmodel lifecycle is invoked. I think that this could provide a lot of power and flexibility (along with the ability to specify a renderer per dialog instance, in another issue), making the dialog plugin useful for a variety of dynamic content creation options.

Thoughts?

**Comment by EisenbergEffect** [comment]
I'm having a hard to following this. Can you provide some consumer code that the developer would want to write and explain the use case more?

**Comment by EisenbergEffect** [comment]
@jedd-ahyoung Can this be closed?
